internal/web/server.go

```diff
@@ -428,30 +375,117 @@ func (s *server) packages(w http.ResponseWriter, r *http.Request) {
-		// TODO have a look at handling latestVersion as return value from DescribePackage again – seems weird
-		latestVersion, err = repo.GetLatestVersion("", pkgName)
-		if err != nil {
-			err = fmt.Errorf("An error occurred fetching latest version of %v: %w\n", pkgName, err)
-			fmt.Fprintf(os.Stderr, "%v\n", err)
+	}
+
+	var idx repo.PackageIndex
+	if err := repo.FetchPackageIndex("", pkgName, &idx); err != nil {
+		s.respondAlertAndLog(w, err, "An error occurred fetching versions of "+pkgName)
+		return
+	}
+	if selectedVersion == "" {
+		selectedVersion = idx.LatestVersion
+	}
+	if selectedVersion != idx.LatestVersion {
+		var mf v1alpha1.PackageManifest
+		if err := repo.FetchPackageManifest("", pkgName, selectedVersion, &mf); err != nil {
+			s.respondAlertAndLog(w, err, fmt.Sprintf("An error occurred fetching manifest of %v in version %v", pkgName, selectedVersion))
+			return
 		}
+		manifest = &mf
+	}
+
+	res, err := s.dependencyMgr.Validate(r.Context(), manifest, selectedVersion)
+	if err != nil {
+		s.respondAlertAndLog(w, err, fmt.Sprintf("An error occurred validating dependencies of %v in version %v", pkgName, selectedVersion))
+		return
 	}
 	err = pkgPageTmpl.Execute(w, &map[string]any{
-		"Error":           err,
-		"CurrentContext":  s.rawConfig.CurrentContext,
-		"Package":         pkg,
-		"Status":          status,
-		"Manifest":        manifest,
-		"LatestVersion":   latestVersion,
-		"UpdateAvailable": pkg != nil && s.isUpdateAvailable(r.Context(), pkgName),
+		"Error":             err,
+		"CurrentContext":    s.rawConfig.CurrentContext,
+		"Package":           pkg,
+		"Status":            status,
+		"Manifest":          manifest,
+		"LatestVersion":     idx.LatestVersion,
+		"UpdateAvailable":   pkg != nil && s.isUpdateAvailable(r.Context(), pkgName),
+		"ValidationResult":  res,
+		"ShowConflicts":     res.Status == dependency.ValidationResultStatusConflict,
+		"SelectedVersion":   selectedVersion,
+		"PackageIndex":      &idx,
+		"ShowConfiguration": (pkg != nil && len(manifest.ValueDefinitions) > 0 && pkg.DeletionTimestamp.IsZero()) || pkg == nil,
 	})
 	checkTmplError(err, fmt.Sprintf("package-detail (%s)", pkgName))
 }
 
+func (s *server) packageConfiguration(w http.ResponseWriter, r *http.Request) {
+	pkgName := mux.Vars(r)["pkgName"]
+	selectedVersion := r.FormValue("selectedVersion")
+	enableAutoUpdate := r.FormValue("enableAutoUpdate")
+	pkg, _, manifest, _, err := describe.DescribePackage(r.Context(), pkgName)
+	if err != nil {
+		s.respondAlertAndLog(w, err, fmt.Sprintf("An error occurred fetching package details of %v", pkgName))
+		return
+	} else if pkg == nil {
+		var mf v1alpha1.PackageManifest
+		if err := repo.FetchPackageManifest("", pkgName, selectedVersion, &mf); err != nil {
+			s.respondAlertAndLog(w, err, fmt.Sprintf("An error occurred fetching manifest of %v in version %v", pkgName, selectedVersion))
+			return
+		}
+		manifest = &mf
+	}
+
+	if values, err := extractValues(r, manifest); err != nil {
```

Comment:
Would you just show a warning, or cancel the operation (i.e. do not update) and then show an error on the UI?

I initially also thought about resolving and even validating the resolved values, but I thought it was not worth it because a configuration can become broken anytime anyway. However I have created a follow up ticket #496 where we could define such behavior overall.